### ISSUE TITLE
"queue.c:1442 (xQueueGenericReceive)- assert failed!"

### ISSUE THREAD

**Issue by GeorgeFlorian** [marcmerlin/SmartMatrix_GFX#7]
**TL;DR:**  
Similar code and project don't work.

I have finished a project using your LED libraries some months ago and it works wonders.  
Today I have tried starting another project and I copied bits and parts of the previous project but this one doesn't work. It returns :
```
Starting SmartMatrix Mallocs
Heap/32-bit Memory Available: 308148 bytes total, 113792 bytes largest free block
8-bit/DMA Memory Available  : 255608 bytes total, 113792 bytes largest free block
SmartMatrix Layers Allocated from Heap:
Heap/32-bit Memory Available: 294344 bytes total, 113792 bytes largest free block
Starting SmartMatrix DMA Mallocs
sizeof framestruct: 00004000
DMA Memory Available before ptr1 alloc: 208588 bytes total, 113792 bytes largest free block
matrixUpdateFrames[0] pointer: 3FFC960C
DMA Memory Available before ptr2 alloc: 208588 bytes total, 113792 bytes largest free block
matrixUpdateFrames[1] pointer: 3FFCD61C
Frame Structs Allocated from Heap:
Heap/32-bit Memory Available: 261128 bytes total, 113792 bytes largest free block
8-bit/DMA Memory Available  : 208588 bytes total, 113792 bytes largest free block
Allocating refresh buffer:
lsbMsbTransitionBit of 0 requires 24576 RAM, 113792 available, leaving 89216 free: 
Raised lsbMsbTransitionBit to 0/7 to fit in RAM
lsbMsbTransitionBit of 0 gives 76 Hz refresh, 120 requested: 
lsbMsbTransitionBit of 1 gives 152 Hz refresh, 120 requested: 
Raised lsbMsbTransitionBit to 1/7 to meet minimum refresh rate
Descriptors for lsbMsbTransitionBit 1/7 with 8 rows require 12288 bytes of DMA RAM
SmartMatrix Mallocs Complete
Heap/32-bit Memory Available: 248808 bytes total, 113792 bytes largest free block
8-bit/DMA Memory Available  : 196268 bytes total, 113792 bytes largest free block
Setting up parallel I2S bus at I2S1
Framebuffer_GFX::begin Width: 64 Height: 32 Num Pixels: 2048
/home/runner/work/esp32-arduino-lib-builder/esp32-arduino-lib-builder/esp-idf/components/freertos/queue.c:1442 (xQueueGenericReceive)- assert failed!
abort() was called at PC 0x40088a51 on core 1

Backtrace: 0x4008c804:0x3ffb1d30 0x4008ca35:0x3ffb1d50 0x40088a51:0x3ffb1d70 0x4013b4ae:0x3ffb1db0 0x4013b78a:0x3ffb1dd0 0x4012b228:0x3ffb1df0 0x4012b291:0x3ffb1e10 0x4012c85c:0x3ffb1e30 0x4012c948:0x3ffb1e60 0x40149513:0x3ffb1e80 0x400dda96:0x3ffb1ed0 0x400d52b1:0x3ffb1ef0 0x400e2177:0x3ffb1fb0 0x40088d65:0x3ffb1fd0
```
```
Decoding stack results
0x40088a51: xTaskIncrementTick at /Users/ficeto/Desktop/ESP32/ESP32/esp-idf-public/components/freertos/tasks.c line 2514
0x400dda96: spiffs_phys_rd at /Users/ficeto/Desktop/ESP32/ESP32/esp-idf-public/components/spiffs/spiffs/src/spiffs_cache.c line 161
0x400d52b1: seekdir at /Users/ficeto/Desktop/ESP32/ESP32/esp-idf-public/components/vfs/vfs.c line 608
0x400e2177: _svfprintf_r at ../../../.././newlib/libc/stdio/vfprintf.c line 1399
0x40088d65: vTaskPlaceOnEventList at /Users/ficeto/Desktop/ESP32/ESP32/esp-idf-public/components/freertos/tasks.c line 2962
```

I've followed the code line by line and side by side and there are no modifications to the Matrix code. I am even using the same ESP32.

I don't know if posting the code is a good idea because there are over 2000 lines of code and it has pretty much the same structure.

```
// ========================== CONFIG START ===================================================

/// SmartMatrix Defines
#define COLOR_DEPTH 24                  // known working: 24, 48 - If the sketch uses type `rgb24` directly, COLOR_DEPTH must be 24
#define kMatrixWidth  64
#define kMatrixHeight 32
const uint8_t kRefreshDepth = 24;       // known working: 24, 36, 48
const uint8_t kDmaBufferRows = 2;       // known working: 2-4, use 2 to save memory, more to keep from dropping frames and automatically lowering refresh rate
const uint8_t kPanelType = SMARTMATRIX_HUB75_32ROW_64COL_MOD8SCAN_P5_2727;
const uint8_t kMatrixOptions = (SMARTMATRIX_OPTIONS_NONE);      // see http://docs.pixelmatix.com/SmartMatrix for options
const uint8_t kBackgroundLayerOptions = (SM_BACKGROUND_OPTIONS_NONE);

SMARTMATRIX_ALLOCATE_BUFFERS(matrixLayer, kMatrixWidth, kMatrixHeight, kRefreshDepth, kDmaBufferRows, kPanelType, kMatrixOptions);
SMARTMATRIX_ALLOCATE_BACKGROUND_LAYER(backgroundLayer, kMatrixWidth, kMatrixHeight, COLOR_DEPTH, kBackgroundLayerOptions);

// const int defaultBrightness = (100*255)/100;        // full (100%) brightness
const int defaultBrightness = (5*255)/100;       // dim: 5% brightness

#define mw kMatrixWidth
#define mh kMatrixHeight
#define NUMMATRIX (kMatrixWidth*kMatrixHeight)

CRGB leds[kMatrixWidth*kMatrixHeight];

// Sadly this callback function must be copied around with this init code
void show_callback() {
    for (uint16_t y=0; y<kMatrixHeight; y++) {
	for (uint16_t x=0; x<kMatrixWidth; x++) {
	    CRGB led = leds[x + kMatrixWidth*y];
	    // rgb24 defined in MatrixComnon.h
	    backgroundLayer.drawPixel(x, y, { led.r, led.g, led.b } );
	}
    }
    // This should be zero copy
    // that said, copy or no copy is about the same speed in the end.
    backgroundLayer.swapBuffers(false);
}

SmartMatrix_GFX *matrix = new SmartMatrix_GFX(leds, mw, mh, show_callback);

// ========================== CONFIG END ======================================================
```
```
void setup() {
  enableCore1WDT();
  esp_wifi_set_ps (WIFI_PS_NONE);
  Serial.begin(115200);
  WiFi.onEvent(WiFiEvent);
  delay(100);

  matrixLayer.addLayer(&backgroundLayer); 
  matrixLayer.begin();
  matrixLayer.setBrightness(defaultBrightness);
  matrixLayer.setRefreshRate(240);
  backgroundLayer.enableColorCorrection(true);
  backgroundLayer.fillScreen( {0x00, 0x00, 0x00} );
  backgroundLayer.swapBuffers();
  matrix->begin();
  delay(100);

  if (!SPIFFS.begin(true)) {
    Serial.println("An Error has occurred while mounting SPIFFS ! Formatting in progress");
    return;
  }
//...
}
```
I am using the latest versions of the libraries:
```

#libraries
lib_deps =  https://github.com/me-no-dev/ESPAsyncWebServer.git
            https://github.com/me-no-dev/AsyncTCP.git
            https://github.com/marcmerlin/SmartMatrix_GFX.git
            https://github.com/marcmerlin/Framebuffer_GFX.git
            https://github.com/adafruit/Adafruit-GFX-Library.git
            https://github.com/FastLED/FastLED.git
```

Do you have any idea what can cause this issue ? Have you encountered this ? It's getting frustrating having basically the same code on the same ESP that refuses to work.

**Comment by GeorgeFlorian** [comment]
It was a SPIFFS error. Sorry to bother you.

**Comment by marcmerlin** [comment]
I was going to say, you did an exceptional job reporting a bug with detailed information, that showed the crash in SPIFFS.
Did you find out what the issue in SPIFFS, was?

**Comment by GeorgeFlorian** [comment]
> I was going to say, you did an exceptional job reporting a bug with detailed information, that showed the crash in SPIFFS.
> Did you find out what the issue in SPIFFS, was?

I don't really know. The ESP already had some files on it, but the project's code was searching for other files. Uploading the proper files to the ESP and as such removing the old ones fixed the issue.

Now, I thought that I handled file not found pretty good but it seems I don't and neither does SPIFFS. It should've just bounced back with a -1 when it didn't find the file or something like that. Instead it was crashing like crazy.
